src/main/java/pdi/project/com/example/recipeapi/repository/UnitMeasureRepository.java

```diff
@@ -0,0 +1,34 @@
+package pdi.project.com.example.recipeapi.repository;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+import pdi.project.com.example.recipeapi.domain.UnitOfMensurement;
+
+@Component
+@Scope("singleton")
+public class UnitMeasureRepository {
+
+  private final List<UnitOfMensurement> units;
+
+  private final UnitOfMensurement kilo = new UnitOfMensurement(1L, "kg");
+  private final UnitOfMensurement unidade = new UnitOfMensurement(2L, "unidade");
+  private final UnitOfMensurement colherDeSopa = new UnitOfMensurement(3L, "colher de sopa");
+
+  public UnitMeasureRepository(List<UnitOfMensurement> units) {
+    this.units = units;
+
+    units.add(kilo);
+    units.add(unidade);
+    units.add(colherDeSopa);
```

Comment:
nas versões mais novas do Java, temos o List.of(), que nos permite criar uma lista de elementos. Nesse caso, poderíamos simplificar a adição de vários items de uma só vez, como `units.addAll(List.of(kilo, unidade, colherDeSopa))`